client/lib/fifo/fifo_windows.go

```diff
@@ -64,6 +65,11 @@ func (f *winFIFO) Write(p []byte) (n int, err error) {
 }
 
 func (f *winFIFO) Close() error {
+	f.connLock.Lock()
+	if f.conn != nil {
+		f.conn.Close()
+	}
+	f.connLock.Unlock()
```

Comment:
How would the lock interact with a Read call?   A Read call will block while holding the `f.connLock` which would then block `Close()`.

Curious - why not initialize `f.conn` in the open functions?  It would avoid the lock and checking `f.conn` in Read and `Write`?

Reply:
Waiting for an Accept in the `Open` call would block indefinitely during some more-edgier cases. It should be safe to reduce the scope of the lock however, to make this a little bit better if we don't have any parallel read/writes going on.

Reply:
FWIW, the linux fifo implementation also blocks indefinitely, as logmon reader blocks until the writer opens.  As such, logmon ensures that the open happens in non-main goroutine.  But yeah, having a finer scoped lock would be ok too.